engine/src/test/java/org/hibernate/validator/test/predefinedscope/PredefinedScopeValidatorFactoryTest.java

```diff
@@ -190,11 +191,26 @@ public void testUninitializedLocale() {
 		}
 	}
 
-	@Test(expectedExceptions = ValidationException.class, expectedExceptionsMessageRegExp = "HV000249:.*")
+	@Test
 	public void testBeanMetaDataClassNormalizerNoNormalizer() {
-		Validator validator = getValidator();
+		// In this case, as we haven't registered any metadata for the hierarchy, even if we have constraints,
+		// we won't have any violations.
+		Set<ConstraintViolation<Bean>> violations = getValidator().validate( new BeanProxy() );
+		assertNoViolations( violations );
 
-		validator.validate( new BeanProxy() );
+		// Now let's register the metadata for Bean and see how it goes
+		Set<Class<?>> beanMetaDataToInitialize = new HashSet<>();
+		beanMetaDataToInitialize.add( Bean.class );
+
+		ValidatorFactory validatorFactory = Validation.byProvider( PredefinedScopeHibernateValidator.class )
+				.configure()
+				.initializeBeanMetaData( beanMetaDataToInitialize )
+				.initializeLocales( Collections.singleton( Locale.ENGLISH ) )
+				.buildValidatorFactory();
+
+		// As we don't have any metadata for BeanProxy, we consider it is not constrained at all.
```

Comment:
that's interesting. So in case something will create a proxy that we don't know about everything would "break". Maybe instead of storing the "final" metadata we could create "descriptors" for the beans that were requested for initialization and then create "final" metadata based only on them ?

Reply:
basically instead of predefined metadata manager, have predefined metadata provider ... not sure it would work though

Reply:
Yes, I agree it's not ideal and that throwing an exception was more comfortable.

I thought about this a bit and I think that would be a significant amount of work as we would need to check all the operations we do when we assemble the metadata and see if they are doing reflection operations.

It's probably something we should do at some point.

Reply:
yeah I agree that it'll take much time. So I think we should be ok as it is now. This is only going into the alpha, right? so we might still have time to do it differently till the final version :) or at least try and see if it's possible..

Reply:
@marko-bekhta I thought a bit about it. I don't think we can reuse the standard path of merging metadata as I'm pretty sure it requires reflection (for instance to determine if a method overrides another) and we don't want any at this point.

What I was wondering is if we should scan the class hierarchy and take into account the first class with existing metadata and then forge a valid metadata instance with the hypothesis that the current class doesn't have any added constraint. I'm not entirely sure it's doable to be honest as I think we will still have the issue with potentially overriding methods. Even if unconstrained, we will have to determine if they override a constrained method.

Reply:
yeah.. the behavior of overridden methods would become unpredictable. What bothered me was that we just "silently" make a bean "unconstrained". I'm not sure that it's worth to make a config knob to fail on uninitialized bean. But maybe let's at least keep a log message instead of removed `throw LOG.uninitializedBeanMetaData( beanClass );` . It'll make one log entry when initializing a metadata for an "unexpected" bean.

Reply:
Unfortunately, I don't think that would fly. People would have a lot of warnings if they validate unconstrained beans.

I think we need to have faith in the implementation as I don't see any other satisfying solution.

Reply:
That's right.... And if we make it a debug, nobody will see it anyway :smile: 

Can't think of anything else we can do here. Should I merge it in or are we waiting on something else ?

Reply:
Hum. Apparently, I already merged it :). Closing then.